### ISSUE TITLE
Missing `href` property on a clicked item

### ISSUE THREAD

**Issue by pkissman** [running-coder/jquery-typeahead#383]
I have a fairly straightforward typeahead that defines an item.href  with a {{display}} placeholder.
Since moving from 2.9.0 to 2.10.3,  the href can no longer be found (is undefined).   Even whey I put in a dummy href, such as "/beer/" it cannot be found.     It seem like the same behavior I am seeing on your country v.2 demo.

Is there a change from 2.9.x that might affect the visibility of the href option?

Here is my code:
It is the same href template for all groups, so not specified at the group level.
```javascript
$.typeahead({
            input: ".js-typeahead",
            minLength: 1,
            order: "asc",
            group: true,
            maxItemPerGroup: 3,
            hint: true,
            dropdownFilter: "All Types",
            href: "/resource-sharing/databases/links.php?locid={{display}}",
            template: "{{display}}",
            emptyTemplate: "no result for {{query}}",
            source: {
                public: {
                    data: data.publics
                },
                academic: {
                    data: data.academics
                },
                school: { 
                    data: data.schools
                },
                special: {
                    data: data.specials
                }
            },
            callback: {
                onClickAfter: function (node, a, item, event) {
                    event.preventDefault();
                    var str = item.href;
                    **alert(str);**
                    var re = /(^.*=).*\[(.*)\]$/i;
                    var newhref = str.replace(re, "$1$2");
                    $(location).attr('href',newhref);
                },
                onResult: function (node, query, obj, objCount) {
                    console.log(objCount)
                    var text = "";
                    if (query !== "") {
                        text = objCount + ' elements matching "' + query + '"';
                    }
                    $('.js-result-container').text(text);
                }
            },
            debug: true
        });
```

**Comment by pkissman** [comment]
I think I fixed this myself.   The {{display}} in these two lines no longer seem to work.
```
href: "/resource-sharing/databases/links.php?locid={{display}}",
template: "{{display}}",

```
So I created my page redirect  using an href function

```
href: function (item) {
                // Example of item.display: Blah Public Library [blah_id]
                var re = /\[(.*)\]$/i;
                var locid = re.exec(item.display); 
                return "/resource-sharing/databases/links.php?locid="+locid[1]
            },
```

```
callback: {
                onClickAfter: function (node, a, item, event) {
                    event.preventDefault;
                    $(location).attr('href',item.href);
                },
```
Not sure if the old way that I did things, the **template: {{display}}** method, should still work or not, so I am not closing.

Thanks

**Comment by tombertrand** [comment]
hey @pkissman I'm looking on the country_v2 demo and it's still working

![image](https://user-images.githubusercontent.com/19352322/31634886-f46e3b24-b292-11e7-8e4a-419a3aa57292.png)

http://www.runningcoder.org/jquerytypeahead/demo/#form-country_v2

could you provide a sample of `data.academics`?
either you need your data to be something like`["item1", "item2", "item3"]` or `[{display: "item1"}]` if you want to be using the `{{display}}` key. Else reach for any`{{objectKey}}` inside your data

**Comment by pkissman** [comment]
Tom,

Here is an example of the first two lines of my data:academic array

academics: [
    "American International College, James J. Shea Sr. Memorial Library [mlin_w_american]",
    "Amherst College, Robert Frost Library [mlin_w_amhercol]"
        ]

Again,  nothing on my end had changed between 2.9 and 2.10.3, but the old code ceased to function.

As for the Country 2 demo,  I now see the Wikipedia url showing below. Thanks.  I had actually clicked on the choice (e.g. Cambodia) and saw an alert box saying “You will be redirected to: undefined. Continue?”  which comes from the item.href in onClickAfter.  That was what I thought was analogous to what I had been seeing.

Thanks.

**Comment by running-coder** [comment]
sorry for the confusion @pkissman indeed the `href` was missing in the newer versions. I've re-added it and a test on top so it does not happen again. Let me know if it works as expected.

Thanks

**Comment by pkissman** [comment]
Thanks @tombertrand.  I tried your develop branch  and the item.href fix worked.

**Comment by tombertrand** [comment]
Merged to master as `2.10.4` release let me know if you encounter any other issue 👍
